src/StatusQ/Controls/StatusBaseButton.qml

```diff
@@ -16,35 +24,26 @@ Rectangle {
         Danger
     }
 
-    property StatusIconSettings icon: StatusIconSettings {
+    property StatusIconSettings iconSettings: StatusIconSettings {
```

Comment:
This and the move to `Button` introduces a breaking change which we should avoid for now. I could see that this was probably done to be consistent with how icon settings work with `StatusMenuItem`. There however, we had to use `iconSettings` over `icon` because the built-in `icon` property couldn't be mutated.

I'm very fine with moving to `iconSettings` altogether in the long run but doubt that this breaking change is needed for this PR.

Maybe you can shine some light onto this :D I'm probably missing something

Reply:
see above regarding the background